engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```
@@ -172,78 +232,211 @@ private void validate(SnapshotInfo snapshotInfo) {
 
-            copyCommand.setOptions(srcDetails);
+            String value = _configDao.getValue(Config.PrimaryStorageDownloadWait.toString());
+            int primaryStorageDownloadWait = NumbersUtil.parseInt(value, Integer.parseInt(Config.PrimaryStorageDownloadWait.getDefaultValue()));
+            CopyCommand copyCommand = new CopyCommand(snapshotInfo.getTO(), templateInfo.getTO(), primaryStorageDownloadWait, VirtualMachineManager.ExecuteInSequence.value());
+
+            String errMsg = null;
+
+            CopyCmdAnswer copyCmdAnswer = null;
 
-            copyCmdAnswer = (CopyCmdAnswer)_agentMgr.send(hostVO.getId(), copyCommand);
-        }
-        catch (Exception ex) {
-            throw new CloudRuntimeException(ex.getMessage());
-        }
-        finally {
             try {
-                _volumeService.revokeAccess(snapshotInfo, hostVO, srcDataStore);
+                // If we are using a back-end snapshot, then we should still have access to it from the hosts in the cluster that hostVO is in
+                // (because we passed in true as the third parameter to createVolumeFromSnapshot above).
+                if (usingBackendSnapshot == false) {
+                    _volumeService.grantAccess(snapshotInfo, hostVO, srcDataStore);
+                }
+
+                Map<String, String> srcDetails = getSnapshotDetails(snapshotInfo);
+
+                copyCommand.setOptions(srcDetails);
+
+                copyCmdAnswer = (CopyCmdAnswer)_agentMgr.send(hostVO.getId(), copyCommand);
             }
             catch (Exception ex) {
-                s_logger.debug(ex.getMessage(), ex);
+                throw new CloudRuntimeException(ex.getMessage());
             }
+            finally {
+                try {
+                    _volumeService.revokeAccess(snapshotInfo, hostVO, srcDataStore);
+                }
+                catch (Exception ex) {
+                    s_logger.debug(ex.getMessage(), ex);
```

Why not log the ignored exception to ``WARN``?